Player.java

```diff
@@ -35,7 +31,7 @@ public boolean isAlive() {
     public void lvlup(){
         while(this.xp>=this.xp_per_level){
             this.xp-=this.xp_per_level;
-            this.hp+=this.hp_increase_per_level;
+            this.hp=this.maxHP+this.hp_increase_per_level;
```

Comment:
en nem ertem, ezt a logikat miert ilyen nehez leprogramozni. Ha most ezt csinaljuk, akkor a hp-t felhuzza maxHP-ra, meg hozzaadja az increase-t. De a jobboldal ugyenannyi lesz az elso meg a masodiuk, meg a 25. szintlepes utan is.

Reply:
Lehet a mukodest nem erted.
Kezdetben monjduk 10 max HP, az increase mondjuk 2. 
Csatazgatunk, ez lemegy 5-re. Szintet lepunk, akkor 12 lesz a maxHP meg az aktualis HP is. csatazgatunk, lemegy 1-re. Szintet lepunk, akkor 14 lesz a maxHP meg az aktualis HP is.

Reply:
A te valtozatodban a maxHP marad mindig 10, es mindig 12-re allitodik be az aktualis hp szintlepeskor.